queries.sql

```diff
@@ -7,6 +7,26 @@ SELECT name, date_of_birth FROM animals WHERE name = 'Agumon' OR name='Pikachu';
 SELECT name, escape_attempts FROM animals WHERE weight_kg >= 10.5;
  SELECT name, neutered FROM animals WHERE neute
 red = TRUE;
-SELECT name FROM animaltests WHERE name <> 'Gabumon';
+SELECT name FROM animals WHERE name <> 'Gabumon';
 SELECT name, weight_kg FROM animals WHERE weig
 ht_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg = weight_kg * (-1);
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg = weight_kg * (-1) WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered,AVG(escape_attempts) FROM animals GROUP BY neutered;
+SELECT neutered, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY neutered;
+SELECT neutered, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY neutered;
```

Comment:
- [ ] These last two queries asked for animal type that is `species` of animal not neutered okay. Kindly select and group by `species` instead of `neutered`.

- [ ] The last query should be `WHERE date_of_birth BETWEEN 1990-01-01 AND 2000-31-12`  so that you get all the average number of escape attempts per animal type of those born between 1990 and 2000?